src/arcsf/eval/metrics.py

```diff
@@ -0,0 +1,98 @@
+import torch
+from rouge_score.rouge_scorer import RougeScorer
+from scipy.stats import ks_2samp
+
+# scorer so it doesn't need to be initialised on every call
+scorer = RougeScorer(["rouge1", "rougeL"], use_stemmer=True)
+
+
+def ks_test(forget: torch.Tensor, retain: torch.Tensor, **kwargs) -> float:
+    """KS test figures 4-6 in the paper
+
+    Args:
+        forget : forget truth ratios
+        retain : retain truth ratios
+
+    Returns:
+        p_value: returns the p_value of the ks_test for use in the forget quality
+    """
+    return ks_2samp(forget, retain, **kwargs).pvalue
+
+
+def eval_accuracy(logits: torch.Tensor, labels: torch.Tensor) -> dict[torch.Tensor]:
+    """Gets the accuracy given logits, this isn't used in the paper.
+
+    Args:
+        logits : output logits
+        labels : targets
+
+    Returns:
+         accuracy : accuracy metric in the form of a dictionary
+    """
+    preds = logits.argmax(dim=-1)
+    accuracy = torch.mean(torch.eq(preds, labels).float())
+    return {"eval_accuracy": accuracy}
+
+
+def conditional_probability(
+    token_normalised_losses: torch.Tensor,
+) -> dict[torch.Tensor]:
+    """
+    Conditional probabilty defined in section 2.2.2 of the paper. Probabilities
+    calculated using:
+    p_i = exp(-1 * p_i) / (sum(exp(-1*p_j)) for j in N)
+
+    Args:
+        token_normalised_losses : losses on the answer normalised by number of tokens
+            shape: n_samples x n_perturbed
+
+    Returns:
+        conditional_probs: conditional probabilities in the form of a dictionary
+    """
+    probs = torch.exp(-1 * token_normalised_losses)  # shape : n_samples x n_perturbed
+    sum = torch.sum(probs, dim=-1)  # shape : n_samples
+    # transpose here ensures that the n_samples dimension is in the right place
+    cond_probs = probs.T / sum  # shape : n_perturbed x n_samples
+    return {"conditional_probs": cond_probs}
```

Comment:
why not just return `cond_probs` here? The only place I can see that this function is used the values are immediately extracted from the dictionary again anyway.

Reply:
I kept this since the original TOFU code did it, I wasn't sure why they used a dictionary either but I thought there might have been some logging tool using it or something. 

Anyway.. I can change it since it isn't needed.